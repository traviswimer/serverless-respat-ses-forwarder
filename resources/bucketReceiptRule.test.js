const bucketReceiptRule = require('./bucketReceiptRule');

describe('bucketReceiptRule()', () => {
	test('generates receipt rule', () => {
		expect(bucketReceiptRule({
			prefix: "ThePrefix",
			pattern_name: "ThePatternName",
			email_recipients: ["some_email@domain"],
			bucket_name: "TheBucketName",
			ses_ruleset_name: "TheRulesetName"
		})).toMatchSnapshot();
	});
});
