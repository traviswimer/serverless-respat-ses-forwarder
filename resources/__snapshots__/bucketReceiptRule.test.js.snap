// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucketReceiptRule() generates policy 1`] = `
Object {
  "DependsOn": Array [
    "ThePatternNameEmailBucket",
  ],
  "Properties": Object {
    "Rule": Object {
      "Actions": Array [
        Object {
          "S3Action": Object {
            "BucketName": "TheBucketName",
          },
        },
      ],
      "Enabled": true,
      "Name": "ThePrefix-ThePatternName-s3-email-store",
      "Recipients": Array [
        "some_email@domain",
      ],
      "ScanEnabled": true,
    },
    "RuleSetName": "TheRulesetName",
  },
  "Type": "AWS::SES::ReceiptRule",
}
`;

exports[`bucketReceiptRule() generates receipt rule 1`] = `
Object {
  "DependsOn": Array [
    "ThePatternNameEmailBucket",
  ],
  "Properties": Object {
    "Rule": Object {
      "Actions": Array [
        Object {
          "S3Action": Object {
            "BucketName": "TheBucketName",
          },
        },
      ],
      "Enabled": true,
      "Name": "ThePrefix-ThePatternName-s3-email-store",
      "Recipients": Array [
        "some_email@domain",
      ],
      "ScanEnabled": true,
    },
    "RuleSetName": "TheRulesetName",
  },
  "Type": "AWS::SES::ReceiptRule",
}
`;
